src/components/settings/EnvironmentVariables/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect } from 'react'
+import { useForm, FormProvider } from 'react-hook-form'
+import { Paper, Grid, Typography, TextField, Button, Tooltip, IconButton } from '@mui/material'
+import InputAdornment from '@mui/material/InputAdornment'
+import RotateLeftIcon from '@mui/icons-material/RotateLeft'
+import { useAppDispatch, useAppSelector } from '@/store'
+import { selectSettings, setCgw } from '@/store/settingsSlice'
+import { GATEWAY_URL_PRODUCTION } from '@/config/constants'
+
+export enum EnvVariablesField {
+  cgw = 'cgw',
+}
+
+export type EnvVariablesFormData = {
+  [EnvVariablesField.cgw]: string
+}
+
+const EnvironmentVariables = () => {
+  const settings = useAppSelector(selectSettings)
+  const dispatch = useAppDispatch()
+
+  const formMethods = useForm<EnvVariablesFormData>({
+    mode: 'onChange',
+    defaultValues: {
+      [EnvVariablesField.cgw]: settings.env.cgw ?? '',
+    },
+  })
+
+  const { register, handleSubmit, reset, setValue, watch } = formMethods
+
+  const cgw = watch(EnvVariablesField.cgw)
+
+  const onSubmit = handleSubmit((data) => {
+    dispatch(setCgw(data[EnvVariablesField.cgw]))
+  })
+
+  const onResetCgw = () => {
+    setValue(EnvVariablesField.cgw, '')
+  }
+
+  useEffect(() => {
+    reset({ [EnvVariablesField.cgw]: settings.env.cgw })
+  }, [reset, settings.env.cgw])
+
+  return (
+    <Paper sx={{ padding: 4 }}>
+      <Grid container direction="row" justifyContent="space-between" spacing={3} mb={2}>
+        <Grid item lg={4} xs={12}>
+          <Typography variant="h4" fontWeight={700}>
+            Environment Variables
+          </Typography>
+        </Grid>
+
+        <Grid item xs>
+          <Typography mb={3}>Only change these if you know what you are doing.</Typography>
```

Comment:
Please ask chatgpt to rewrite this in a non-arrogant style :)

Reply:
I changed it to say "You can override some of our default APIs here in case you need to. Proceed at your own risk."